transport/src/main/java/io/netty5/channel/DefaultChannelHandlerContext.java

```diff
@@ -934,6 +999,20 @@ private static boolean safeExecute(EventExecutor executor, Runnable runnable, Pr
         }
     }
 
+    @Override
+    public void execute(Runnable task) {
+        executor().execute(() -> {
+            // Ensure we update the pending tasks if needed as the Runnable might execute handler methods directly
+            // which we entered from within this context.
```

Comment:
When we still have the `executor()` method public, this wrapping can be bypassed, with a very unintuitive change in behaviour as a result.

Reply:
I agree .., I was also wondering if we should maybe just wrap the returned EventExecutor here and not add new methods. This might be even better and less error-prone and we can still update the byes under the hood